libs/core/langchain_core/runnables/config.py

```diff
@@ -161,9 +162,10 @@ def ensure_config(config: Optional[RunnableConfig] = None) -> RunnableConfig:
         configurable={},
     )
     if var_config := var_child_runnable_config.get():
-        empty.update(
-            cast(RunnableConfig, {k: v for k, v in var_config.items() if v is not None})
+        deep_copied_config = copy.deepcopy(
+            {k: v for k, v in var_config.items() if v is not None}
         )
+        empty.update(cast(RunnableConfig, deep_copied_config))        
```

Comment:
```suggestion
        empty = merge_configs(empty, {k: v for k, v in var_config.items() if v is not None})
```

Reply:
Hi @nfcampos thanks for the suggestion

If I follow your suggestion, I wonder
1. if I can still achieve the purpose of getting a "safe to mutate" copy of the config ? For example in `"k, v" of var_config.items()` could be the callbacks, and the reader of it can unintended mutation of the fields in it like the "handlers" 
2. if calling the "merge_configs" function in "ensure_config" will lead to infinite recursion? since merge_configs also calls ensure_config

Reply:
hi, on the first point, notice that merge_configs doesn't mutate any of the parts of config it modifies, it create new copies. on the 2nd point, good point, I'll make a version of merge_configs that doesn't call ensure_config